### PR TITLE
Load config from different locations prefer XDG_CONFIG_HOME before HOME

### PR DESCRIPTION
**Pull request by mfederowicz**
base on https://github.com/mgechev/revive/pull/725 

make some cleanup, and tests using github.com/spf13/afero package for mocking filesystem

### CODE REVIEW

**Comment by mfederowicz** [comment]
@chavacava can you look at my changes? Sorry for go.sum mess but I can find solution to make file with latest checksums for packages :(

File: cli/main.go [hunk 1]
```diff
@@ -175,3 +190,12 @@ func init() {
 		os.Exit(0)
 	}
 }
+
+func fileExist(path string) bool {
+	_, err := AppFs.Stat(path)
+	if err != nil {
```
Comment (chavacava) [inline cli/main.go:196]:
```suggestion
	return err == nil 
```

File: cli/main_test.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+import (
+	"os"
+	"testing"
+
+	"github.com/spf13/afero"
+)
+
+func init() {
+
```
Comment (chavacava) [inline cli/main_test.go:11]:
```suggestion
```

File: cli/main_test.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    AppFs = afero.NewMemMapFs()
+}
+
+func TestHomeConfigDir(t *testing.T) {
+
```
Comment (chavacava) [inline cli/main_test.go:16]:
```suggestion
```

File: cli/main_test.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	afero.WriteFile(AppFs, homeDirPath+"/revive.toml", []byte("\n"), 0644)
+	os.Setenv("HOME", homeDirPath)
+	
+    got := buildDefaultConfigPath()
+	want := homeDirPath + "/revive.toml"
+
+	if got != want {
+		t.Errorf("got %q, wanted %q", got, want)
+	}
+
```
Comment (chavacava) [inline cli/main_test.go:29]:
```suggestion
```

File: cli/main_test.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+func TestXDGConfigDir(t *testing.T) {
+
```
Comment (chavacava) [inline cli/main_test.go:33]:
```suggestion
```

File: cli/main_test.go [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+	afero.WriteFile(AppFs, xdgDirPath+"/revive.toml", []byte("\n"), 0644)
+	os.Setenv("XDG_CONFIG_HOME", xdgDirPath)
+
+	got := buildDefaultConfigPath()
+	want := xdgDirPath + "/revive.toml"
+
+	if got != want {
+		t.Errorf("got %q, wanted %q", got, want)
+	}
+
```
Comment (chavacava) [inline cli/main_test.go:46]:
```suggestion
```

File: cli/main_test.go [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+func TestXDGConfigDirNoFile(t *testing.T) {
+
```
Comment (chavacava) [inline cli/main_test.go:50]:
```suggestion
```

Formatting of this function body is not OK

File: lint/config.go [hunk 8]
```diff
@@ -33,8 +33,8 @@ func (rc *RuleConfig) Initialize() error {
 type RulesConfig = map[string]RuleConfig
 
 // MustExclude - checks if given filename `name` must be excluded
-func (rcfg *RuleConfig) MustExclude(name string) bool {
-	for _, exclude := range rcfg.excludeFilters {
+func (rc *RuleConfig) MustExclude(name string) bool {
```
Comment (chavacava) [inline lint/config.go:36]: Thanks! :)

File: cli/main_test.go
Comment (chavacava) [inline cli/main_test.go:1]: We could add a tescase where there are a `revive.toml` in both `$HOME` and `$XDG_CONFIG_HOME` to check we load that of `$XDG_CONFIG_HOME`

**Review by chavacava — COMMENTED** [review]
Hi @mfederowicz, thank you for the PR! 
I've left some comments and minor formatting suggestions.

File: cli/main_test.go
Comment (mfederowicz) [reply cli/main_test.go:1]: hehe i already added that case just fee minute ago :P

**Comment by mfederowicz** [comment]
ok @chavacava I commited changes with yours suggestions, let me know i all good, and then we can squash commits and go to master :P

**Pull request merged**